{
  "name": "InvalidThrows",
  "language": "java",
  "description": "The documented method doesn't actually throw this checked exception.",
  "example": null,
  "cwe": null,
  "cwe-description": null,
  "checker-language": "java",
  "loc": 84,
  "branches": 5,
  "apis": 4,
  "test": [
    {
      "description": "positive",
      "expected-problems": null,
      "expected-linenumbers": [
        31
      ],
      "code": "import java.io.IOException;\n\n            interface Test {\n              /**\n               * @throws IOException when failed\n               */\n              void foo(int a, int b);\n            }\n            \n\n            import java.io.IOException;\n\n            interface Test {\n              /** */\n              void foo(int a, int b);\n            }"
    },
    {
      "description": "negative",
      "expected-problems": null,
      "expected-linenumbers": [
        59
      ],
      "code": "import java.io.IOException;\n\n            interface Test {\n              /**\n               * @throws IOException when failed\n               */\n              void bar(int a, int b) throws IOException;\n\n              /**\n               * @throws IOException when failed\n               */\n              void baz(int a, int b) throws Exception;\n            }"
    },
    {
      "description": "unrecognisedException_noComplaint",
      "expected-problems": null,
      "expected-linenumbers": [
        83
      ],
      "code": "interface Test {\n              /**\n               * @throws com.google.UnknownRuntimeException when failed\n               */\n              void bar(int a, int b);\n            }"
    },
    {
      "description": "throwsInvalidType",
      "expected-problems": null,
      "expected-linenumbers": [
        100
      ],
      "code": "interface Test {\n              /**\n               * @throws bar when failed\n               */\n              void foo();\n\n              void bar();\n            }"
    }
  ]
}